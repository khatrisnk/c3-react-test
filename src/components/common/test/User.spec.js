import React from 'react';
import UserComponent from '../User';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

const userProps = {
  first_name: 'Siddhartha',
  last_name: 'Khatri',
  avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg',
  id: 7,
  deleteUser: jest.fn()
};

describe('UserComponent Sanpshots', () => {
  it('UserComponent Sanpshot - with all the props', () => {
    const component = renderer.create(<UserComponent {...userProps} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('UserComponent', () => {
  it('Simulate click event on delete link', () => {
    const wrapper = mount(<UserComponent {...userProps} />);
    wrapper.find('.user-delete').simulate('click');
    expect(wrapper.props().deleteUser).toHaveBeenCalled();
  });
  it('renders first name', () => {
    const wrapper = shallow(<UserComponent {...userProps} />);
    expect(wrapper.find('.user-first-name')).toHaveLength(1);
  });
  it('renders last name', () => {
    const wrapper = shallow(<UserComponent {...userProps} />);
    expect(wrapper.find('.user-last-name')).toHaveLength(1);
  });
  it('renders avatar', () => {
    const wrapper = shallow(<UserComponent {...userProps} />);
    expect(wrapper.find('.user-avatar')).toHaveLength(1);
  });
});
