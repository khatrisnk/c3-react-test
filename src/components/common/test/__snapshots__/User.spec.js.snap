// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserComponent Sanpshots UserComponent Sanpshot - with all the props 1`] = `
<div
  className="user col-md-4"
>
  <div
    className="user-top"
  />
  <div
    className="user-bottom"
  >
    <div
      className="user-name"
    >
      <span
        className="user-first-name"
      >
        Siddhartha
      </span>
      Â 
      <span
        className="user-last-name"
      >
        Khatri
      </span>
    </div>
    <a
      className="user-delete"
      href="# "
      onClick={[Function]}
      title="delete"
    >
      Delete
    </a>
  </div>
  <img
    alt="avtar"
    className="user-avatar"
    src="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
  />
</div>
`;
